mirascope/core/groq/call_response.py

```diff
@@ -162,3 +166,24 @@ def tool_message_params(
             )
             for tool, output in tools_and_outputs
         ]
+
+    @property
+    def common_finish_reasons(self) -> list[FinishReason] | None:
+        return cast(list[FinishReason], self.finish_reasons)
+
+    @property
+    def common_message_param(self) -> BaseMessageParam:
```

Comment:
realizing that we need to update `BaseMessageParam` to account for tool calls. This means we'll need to add support for `role="tool"` (including whatever other metadata tool messages might have and converting these into provider-specific tool messages, which we currently don't do because we assume the user is using provider-specific tool messages right now).